firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -1697,9 +1697,17 @@ class BrowserViewController: UIViewController,
             // Ensure that the tab title *actually* changed to prevent repeated calls
             // to navigateInTab(tab:).
             guard let title = tab.title else { break }
-            if !title.isEmpty && title != tab.lastTitle {
-                tab.lastTitle = title
-                navigateInTab(tab: tab, webViewStatus: .title)
+            if !title.isEmpty {
+                if title != tab.lastTitle {
+                    tab.lastTitle = title
+                    navigateInTab(tab: tab, webViewStatus: .title)
+                } else {
+                    if let readerMode = tab.getContentScript(name: ReaderMode.name()) as? ReaderMode {
+                        if readerMode.state == .active {
+                            navigateInTab(tab: tab, webViewStatus: .title)
+                        }
+                    }
```

Comment:
nit: what do you think about moving this into a private func with a descriptive name that gives us a better clue on why we need to add this logic.

Reply:
Yes, I will shift it to a private function.